src/lib/y2storage/proposal/autoinst_size_parser.rb

```diff
@@ -0,0 +1,152 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/proposal/autoinst_size"
+require "y2storage/volume_specification"
+
+module Y2Storage
+  module Proposal
+    # This class parses the 'size' AutoYaST element and returns an
+    # {Y2Storage::Proposal::AutoinstSize} object
+    class AutoinstSizeParser
+      # @return [ProposalSettings] Proposal settings
+      attr_reader :proposal_settings
+
+      # Constructor
+      #
+      # @param proposal_settings [ProposalSettings] Proposal settings
+      def initialize(proposal_settings)
+        @proposal_settings = proposal_settings
+      end
+
+      # Return AutoinstSize for a given AutoYaST 'size' element
+      #
+      # @example Parsing 'auto'
+      #   parser = AutoinstSizeParser.new(ProposalSettings.new_for_current_product)
+      #
+      #   size_info = parser.parse("10GiB", "/", 1.GiB, 15.GiB)
+      #   size_info.min #=> 10.GiB
+      #   size_info.max #=> 10.GiB
+      #
+      # @param section [Y2Storage::AutoinstProfile::PartitionSection]
+      # @param min     [DiskSize] Minimal size
+      # @param max     [DiskSize] Maximal size
+      # @return [AutoinstSize] Object containing information about size
+      #
+      # @see AutoinstSize
+      def parse(size_spec, mount_point, min, max)
```

Comment:
Fix doc for parameters: `section` does not exist and `size_spec`, `min` and `max` are missing.